contracts/Core/StakeManager.sol

```diff
@@ -146,10 +146,10 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause {
         uint256 rAmount = _convertSRZRToRZR(sAmount, staker.stake, sToken.totalSupply());
         staker.stake = staker.stake - rAmount;
 
-        locks[msg.sender][staker.tokenAddress] = Structs.Lock(rAmount, epoch + (parameters.withdrawLockPeriod()));
-
         staker.epochLastUnstakedOrFirstStaked = epoch;
 
+        locks[msg.sender][staker.tokenAddress] = Structs.Lock(rAmount, epoch + (parameters.withdrawLockPeriod()));
```

Comment:
Why its position is changed ?

Reply:
before 
```staker.epochLastUnstakedOrFirstStaked = epoch;```
or after 
```
staker.epochLastUnstakedOrFirstStaked = epoch;
```
does it matter ?

Reply:
parameters.withdrawlockPeriod() is external call

Reply:
oh okay